src/lib/mdns/DiscoveryManager.cpp

```diff
@@ -330,5 +340,75 @@ void DiscoveryManager::HandleNodeIdResolve(void * context, MdnsService * result,
 #endif // CHIP_ENABLE_MDNS
 }
 
+void DiscoveryManager::AddMdnsService(const MdnsService & service)
+{
+#if CHIP_ENABLE_MDNS
```

Comment:
could we somehow avoid having #ifdef everywhere?

Could we separate out at build-time either an empty implementation or a full features implementation?